mongodb/src/test/java/org/apache/calcite/adapter/mongodb/MongoAdapterTest.java

```diff
@@ -859,4 +859,32 @@ private static Consumer<List> mongoChecker(final String... expected) {
         .limit(2)
         .returns("STATE=VT; AVG(pop)=26408\nSTATE=AK; AVG(pop)=26856\n");
   }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-2109">[CALCITE-2109]
+   * Mongo adapter: unable to translate (A AND B) conditional case</a>. */
+  @Test void testTranslateAndInCondition() {
+    assertModel(MODEL)
+        .query("select state, city from zips "
+            + "where city='LEWISTON' and state in ('ME', 'VT') "
```

Comment:
How about adding a test case of "A OR B AND C"?

Reply:
@NobiGo sure, done.